code/espurna/sensors/SHT3XI2CSensor.h

```diff
@@ -78,11 +78,43 @@ class SHT3XI2CSensor : public I2CSensor<> {
 
             unsigned char buffer[6];
             i2c_read_buffer(address, buffer, std::size(buffer));
-
-            // cTemp msb, cTemp lsb, cTemp crc, humidity msb, humidity lsb, humidity crc
-            _temperature = ((((buffer[0] * 256.0) + buffer[1]) * 175) / 65535.0) - 45;
-            _humidity = ((((buffer[3] * 256.0) + buffer[4]) * 100) / 65535.0);
-
+            
+            if ((CRC8(buffer[0],buffer[1],buffer[2])) && (CRC8(buffer[3],buffer[4],buffer[5]))) {
+                // cTemp msb, cTemp lsb, cTemp crc, humidity msb, humidity lsb, humidity crc
+                _temperature = ((((buffer[0] * 256.0) + buffer[1]) * 175) / 65535.0) - 45;
+                _humidity = ((((buffer[3] * 256.0) + buffer[4]) * 100) / 65535.0);
+            }
+            else {
+                _error = SENSOR_ERROR_CRC;
+            }
+            status_register();
+        }
+        
+        // Read the status register and output to Debug log
+        void status_register() {
+            const auto address = lockedAddress();
+            
+            unsigned char buffer[3];
+            bool crc, cmd, htr;
+            // Read status register
+            i2c_write_uint8(address, 0xF3, 0x2D);
+            espurna::time::blockingDelay(
+                espurna::duration::Milliseconds(500));
+            i2c_read_buffer(address, buffer, std::size(buffer));
+            if (CRC8(buffer[0],buffer[1],buffer[2])) {
+                // see https://sensirion.com/resource/datasheet/sht3x-d
+                crc = buffer[1] & 0b00000001;
+                cmd = buffer[1] & 0b00000010;
+                htr = buffer[0] & 0b00100000;
+                DEBUG_MSG_P(PSTR("[SHT3X] Status crc:%u cmd:%u htr:%u\n"), crc, cmd, htr);
```

Comment:
is the log here intentional w/o SENSOR DEBUG flag? this happens every read.
plus, `%02X%02X` with buffer msb and lsb so we see the whole number?

Reply:
Used SENSOR_DEBUG flag and added whole status output